### PR TITLE
Window management

### PR DESCRIPTION
**Pull request by drucelweisse**
# Description
This pr contains changes about window Management

## Changes

- Added EditorViewController static func makeWindow(for document: VODesingDocument) that creates new window
- removed use of ProjectsRouter, cause on remaking of new window controller loses it's router and don't open window,

## TODO:

- [X] Open Projects on document close
- [x] Open Projects on dock click

## ScreenShots

https://user-images.githubusercontent.com/36012972/181374113-5f53a2ce-3856-43e0-8d3b-59df66dc11a2.mov

Help needed, to determine what event fired on dock click

### CODE REVIEW

**Comment by akaDuality** [comment]
Probably we can terminate window when all windows close, the projects will open like app's restart.

AppDelegate:
```
public func applicationShouldTerminateAfterLastWindowClosed(_ sender: NSApplication) -> Bool {
   true
}
```

**Comment by drucelweisse** [comment]
Yes, that worked

**Comment by akaDuality** [comment]
Have you done the PR?

File: VoiceOver Designer/Features/Package.swift [hunk 1]
```diff
@@ -42,7 +42,9 @@ let package = Package(
         .target(
             name: "Projects",
             dependencies: [
-                "CommonUI"
+                .product(name: "Document", package: "Document"),
+                "CommonUI",
+                "Editor"
```
Comment (akaDuality) [inline VoiceOver Designer/Features/Package.swift:47]: Projects and Editor are feature-modules and shouldn't depend on each other

File: VoiceOver Designer/Features/Sources/Projects/ProjectsViewController.swift [hunk 2]
```diff
@@ -38,7 +39,8 @@ public class ProjectsViewController: NSViewController {
     
     @objc func createNewProject() {
         let document = VODesignDocument()
-        router?.show(document: document)
```
Comment (akaDuality) [inline VoiceOver Designer/Features/Sources/Projects/ProjectsViewController.swift:41]: Router was created to delegate to main app and remove dependencies between modules

**Review by akaDuality — CHANGES_REQUESTED** [review]
Can we restore router to remove dependencies?

File: VoiceOver Designer/Features/Sources/Projects/ProjectsViewController.swift (hunk 2 above)
Comment (drucelweisse) [reply VoiceOver Designer/Features/Sources/Projects/ProjectsViewController.swift:41]: Yeah, I've described it in pr, but for some reasons, on recreate projectsController it lost reference to WindowController as well router becomes nil.

I'll give it a new try :)

Comment (akaDuality) [reply VoiceOver Designer/Features/Sources/Projects/ProjectsViewController.swift:41]: I've discovered that Editor already depends on Settings. I'm going to fix it too

Comment (akaDuality) [reply VoiceOver Designer/Features/Sources/Projects/ProjectsViewController.swift:41]: We will keep router explicitly in AppDelegate. I can do it before tommorow

Comment (akaDuality) [reply VoiceOver Designer/Features/Sources/Projects/ProjectsViewController.swift:41]: I'll create some sort of main controller in the app's part, will show you a PR

Comment (akaDuality) [reply VoiceOver Designer/Features/Sources/Projects/ProjectsViewController.swift:41]: Check #44

**Comment by akaDuality** [comment]
You can finish this PR without tests

File: VoiceOver Designer/Features/Package.swift (hunk 1 above)
Comment (drucelweisse) [reply VoiceOver Designer/Features/Package.swift]: resolved in https://github.com/VODGroup/VoiceOverDesigner/pull/42/commits/6d3e3129e96dffc2cd16f035efc92122423a7dec

File: VoiceOver Designer/Features/Sources/Projects/ProjectsViewController.swift (hunk 2 above)
Comment (drucelweisse) [reply VoiceOver Designer/Features/Sources/Projects/ProjectsViewController.swift:41]: resolved in https://github.com/VODGroup/VoiceOverDesigner/pull/42/commits/7e994525e931d06a04cac384a7c1647621e50dc3

**Comment by akaDuality** [comment]
@drucelweisse I've rafactor to window delegate. Can you check the result?

**Review by drucelweisse — COMMENTED** [review]
Yeah, looks good and removes call to NSApplication shared instance

**Pull request merged**